src/frame.rs

```diff
@@ -199,25 +200,38 @@ impl Frame {
         FrameResolution { width, height }
     }
 
-    pub fn get_pixel_data(&self) -> Vec<Option<&[u8]>> {
+    pub fn get_pixel_data(&self) -> Result<Vec<Option<&[u8]>>, String> {
```

Comment:
Perhaps we can return an Option instead of a Result, like we do for `descriptor`

In this way we tell a developer that there are no data currently available. If we prefer a Result, we can use [Errors](https://github.com/rust-av/ffms2-rs/blob/master/src/lib.rs#L22) instead of a String. [Here](https://github.com/rust-av/ffms2-rs/blob/423429eece9ebb8b5b879e8bad19c2653e9f5edd/src/index.rs) you can find some examples for its usage

Reply:
Thanks! I'm learning quite a lot from this.

Sorry for being a bit of a noob :)

Reply:
Thanks to you, you're doing great! :) 

No problem!